usr/share/rear/prep/default/320_include_uefi_env.sh

```diff
@@ -47,13 +47,21 @@ if grep -qw efivars /proc/mounts; then
     SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
 fi
 
-# next step, is checking /boot/efi directory (we need it)
-if [[ ! -d /boot/efi ]]; then
-    return    # must be mounted
+# next step, is case-sensitive checking /boot for case-insensitive /efi directory (we need it)
+if [[ -n `find /boot -maxdepth 1 -iname efi -type d` ]]; then
+    return    # not found
 fi
 
 # next step, check filesystem partition type (vfat?)
+local efi_mount_point=""
 UEFI_FS_TYPE=$(awk '/\/boot\/efi/ { print $3 }' /proc/mounts)
+# if not mounted at /boot/efi, try /boot itself
+if [[ -z "$UEFI_FS_TYPE" ]]; then
+    UEFI_FS_TYPE=$(awk '/\/boot/ { print $3 }' /proc/mounts)
+    [[ -z "$UEFI_FS_TYHPE" ]] && efi_mount_point='/\/boot/'
```

Comment:
@gozora Thank you: your reasonning that something is wrong is right. 
`[[ -z "$UEFI_FS_TYPE" ]] && efi_mount_point='/\/boot/'` was intended as the inverse `[[ -z "$UEFI_FS_TYPE" ]] || efi_mount_point='/\/boot/'` (set efi_mount_point when UEFI_FS_TYPE is not empty).
Your fix will work, still doesn't feel right. I will make another adjustment which will might make the code better readable.